File: python/problem012.py

def getPrime(L,max):
	prime=[2,3,5]
	L+=[2,3,5]
	count=3
	candidate=7
	isCandidateConfirmed=True
	while(candidate<max):
		isCandidateConfirmed=True
		i=0
		while(prime[i]**2<=candidate):
			if(candidate%prime[i]==0):
				isCandidateConfirmed=False
				break
			i+=1
		if(isCandidateConfirmed==True):
			prime.append(candidate)
			L.append(candidate)
			count+=1
		candidate+=2

def GetCount(L,number):
	num=number
	#print L
	sum=1
	temp=0
	i=0
	while(i<1000 and L[i]<=number):
		temp=0
		#if(number%2!=0 or number%3!=0 ):
		#	return 0
		while(number%L[i]==0):
			temp+=1
			number=number/L[i]
		if(temp):
			sum*=temp+1
		i+=1
	if(sum>500 and sum <1000):
		print num,":",sum

#function return next triangilar
def NextTriangilar(sum,number):
	return sum+number
import time

t=time.time()

L=[]
getPrime(L,100000)
#print "time",time.time()-t
number=0
sum=0
while(number<13000):
	sum=NextTriangilar(sum,number)
	#GetCount(L,number)
	#if(number==6+1):
	GetCount(L,sum);
	number+=1
#print GetCount(L,28)
#print GetCount(28)
#print "time",time.time()-t

File: java/Problem012/src/Problem012.java
public class Problem012 {
	public static int getDelimiterCount(int source, long primes[]) {
		//primeGenerator.printCurrentSequence();
		

		int count = 0;
		int i = 0;
		int sum = 1;
		while( i != primes.length && primes[i] != 0) {
			
			if(source % primes[i] == 0) {
				count++;
				source /= primes[i];
			}
			else {
				sum *= count + 1;
				count = 0;
				i++;
				if(source == 1)
					break;
			}
		}
		
		//primeGenerator.printCurrentSequence();
		return sum;
		/*
		int count = 2;
		for (int i = 2; i < source / 2 + 1; i++) {
			if (source % i == 0) 
				count++;
		}
		return count;*/
	}
	public static void main(String args[]) {
		
		//StdOut.printf("%d%n", getDelimiterCount(21));
		Prime primeGenerator = new Prime();
		primeGenerator.generateMax(1000000);
		long primes[] = primeGenerator.getCurrentSequence();
		
		TriangleNumbersGenerator generator = new TriangleNumbersGenerator();
		int maxnumber = 0;
		int maxsize = 0;
		for (int i = 1;	; i++) {
			int temp = generator.next();
			int count = getDelimiterCount(temp,primes);
			if (maxsize < count) {
				maxsize = count;
				maxnumber = temp;
			}

			StdOut.printf("Triangle number % 8d have % 4d delimiters \t\t\t", temp, count);
			StdOut.printf("%d - %d%n", maxnumber, maxsize);

			if (count > 500) 
				break;
		}
	
		
	}
}